accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -1461,3 +1461,319 @@ wrapper class is so huge that it is likely to be a major use case for
+specification.
+
+#### Conditions and `bool`
+
+Expressions in condition position (`if (condition) …`, (`do {…}`)`while
+(condition)`, `condition ? … : …`, `condition || condition`, `condition &&
+condition`, `!condition`, `… when condition` ) *must* have a type assignable to
+`bool`.
+
+There is no change to that. An extension type occurring in such a position is a
+compile-time error if it does not implement `bool`., and not if it does
+implement `bool`. All runtime behavior on `bool` values amount to checking
+whether it’s the `true` or the `false` value, which is still sound. _I do not
+believe we make any assumptions about members of `bool`, or invoke any
+explicitly._
+
+#### Numbers and arithmetic operators
+
+We have special rules in type inference for number operations on `int`, `num`
+and `double`, to ensure that the static type of `1 + 1` is `int`. We effectively
+pretend to have overloaded operators for typing only.
+
+The [newest
+version](https://github.com/dart-lang/language/blob/main/accepted/2.12/nnbd/number-operation-typing.md)
+of those rules came with null safety. The way the rules are written, they apply,
+e.g., to `e1 + e2` when the static type of `e1` is a non-`Never` subtype of
+`int` (the receiver is an `int`, maybe typed as a type variable `X` with bound
+`int`), and the static type of `e2` is a subtype of `int`. That needs to also
+say that the `+` must denote an instance member, to avoid applying to an
+extension type implementing `int` and declaring an extension type  `+` operator.
+
+That amounts to two sections being changed by inserting something to the effect
+of “where it’s an instance member”, additions italicized here:
+
+>Let `e` be an expression of one of the forms `e1 + e2`, `e1 - e2`, `e1 * e2`,
+>`e1 % e2` or `e1.remainder(e2)`, where the static type of `e1` is a
+>non-~~`Never`~~_bottom_ type *T* and *T* <: `num`, ~~and~~ where the static
+>type of `e2` is *S* and *S* is assignable to `num`_, and where *T* has an
+>implementation of the operator, or `remainder` method, that is an instance
+>member_. Then:
+
+and
+
+>Let `e` be a normal invocation of the form `e1.clamp(e2, e3)`, where the static
+>types of `e1`, `e2` and `e3` are *T*<sub>1</sub>, *T*<sub>2</sub> and
+>*T*<sub>3</sub> respectively, ~~and~~ where *T*<sub>1</sub>, *T*<sub>2</sub>,
+>and *T*<sub>3</sub> are all non-~~`Never`~~_bottom_ subtypes of `num`_, and
+>where *T*<sub>1</sub> has an instance member named `clamp`_ . Then:
+
+This will prevent the rules from applying to invocations of extension type
+operators or methods, which shadow the platform methods that we know and trust.
+
+No other changes should be needed. The remainder of the rules are about context
+types, and require the types involved to be *supertypes* of a number interface
+type, which extension types never are.
+
+#### Patterns
+
+All map pattern entry patterns and list pattern element patterns are define in
```

Comment:
Checking the pattern feature specification, we have no occurrences of 'entry pattern' or 'element pattern' denoting the entire construct (such as `<expression> ':' <pattern>`). It could also be confusing if 'entry pattern' means `<expression> ':' <pattern>` and 'entry subpattern' means just the pattern which is part of the entry. So maybe:

```suggestion
Map pattern entries and list pattern elements have a semantics which is defined in
```